File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.py
while 1:
    num = input()
    if num == '0':
        break
    lst = list(num)
    n = len(lst)
    cnt = 0
    for i in range(n//2):
        if lst[i] == lst[n-1-i]:
            cnt += 1
    if n//2 == cnt:
        print('yes')
    else:
        print('no')
    

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

// 맨 뒤 0 제거
input.pop();

for (let i = 0; i < input.length; i++) {
  let lst = input[i].split("").map(Number);
  let isPelin = true;
  while (lst.length > 1) {
    let first = lst.shift();
    let last = lst.pop();
    if (first != last) {
      isPelin = false;
      break;
    }
  }
  isPelin ? console.log("yes") : console.log("no");
}
